google/cloud/documentai_toolbox/wrappers/document.py

```diff
@@ -147,24 +151,37 @@ def _get_batch_process_metadata(
 
         operation_name (str):
             Required. The fully qualified operation name for a `batch_process_documents()` operation.
+
+        polling_interval (int):
+            Optional. Default is 0. Time in seconds to wait between polls for operation status.
+            Needed to avoid hitting quotas.
+
+        timeout (float):
+            Optional. Default None. Time in seconds to wait for operation to complete.
+            If None, will wait indefinitely.
     Returns:
         documentai.BatchProcessMetadata:
             Metadata from batch process.
     """
     client = documentai.DocumentProcessorServiceClient(
         client_options=ClientOptions(
             api_endpoint=f"{location}-documentai.googleapis.com"
-        )
+        ),
+        client_info=gcs_utilities._get_client_info(),
     )
 
+    # Poll Operation until complete.
     while True:
         operation: Operation = client.get_operation(
-            request=GetOperationRequest(name=operation_name)
+            request=GetOperationRequest(name=operation_name),
+            timeout=timeout,
         )
 
         if operation.done:
             break
 
+        time.sleep(polling_interval)
```

Comment:
Can we use the built-in `retry` here, instead of managing polling ourselves?

https://github.com/googleapis/google-cloud-python/blob/main/packages/google-cloud-documentai/google/cloud/documentai_v1/services/document_processor_service/client.py#L3532

Reply:
That is a much better way to implement, I thought based on the documentation that was only used for errors, not incomplete operations, but I can change it to use that.